app/js/lib/foxbox/settings.js

```diff
@@ -203,4 +187,33 @@ export default class Settings extends EventDispatcher {
   get apiVersion() {
     return API_VERSION;
   }
+
+  clear() {
+    return new Promise((resolve) => {
+      this.session = null;
+      // @todo Remove only the items set here.
```

Comment:
mmm yeah we can, the only thing is that we'll start using these strings in several places, maybe keep them in a single place and use in all related pieces so that we don't misspell and not forget about all places to update when new setting is added or existing is renamed? Like this:
```
const settings = Object.freeze({
  CONFIGURED: { key: 'configured', type: 'boolean' },
  SKIP_DISCOVERY: { key: 'skipDiscovery', type: 'boolean' },
  SERVICE_POLLING_ENABLED: {
    key: 'servicePollingEnabled',
    type: 'boolean',
    defaultValue: DEFAULT_SERVICE_POLLING_ENABLED,
  },
  LOCAL_ORIGIN: { key: 'localOrigin' },
  TUNNEL_ORIGIN: { key: 'tunnelOrigin' },
  CLIENT: { key: 'client' },
  SESSION: { key: 'session' },
  PUSH_ENDPOINT: { key: 'pushEndpoint' },
  PUSH_PUB_KEY: { key: 'pushPubKey' },
});

export default class Settings extends EventDispatcher {
  ......
  init() {
    Object.keys(settings).forEach((settingName) => {
      const setting = settings[settingName];

      let defaultValue = setting.defaultValue;
      if (defaultValue === undefined) {
        defaultValue = setting.type === 'boolean' ? false : null;
      }

      const storedValue = storage.getItem(`${PREFIX}${setting.key}`);

      let value;
      if (storedValue === null) {
        value = defaultValue;
      } else {
        value = setting.type === 'boolean' ?
          storedValue === 'true' : storedValue;
      }

      this[p[setting.key]] = value;
    });
  }

  get [settings.CONFIGURED.key]() {
    return this[p[settings.CONFIGURED.key]];
  }

  set [settings.CONFIGURED.key](value) {
    this[p.updateSetting](settings.CONFIGURED.key, value);
  }

  clear() {
    return new Promise((resolve) => {
      Object.keys(settings).forEach((settingName) => {
        storage.removeItem(`${PREFIX}${settings[settingName].key}`);
      });
      resolve();
    });
  }
```

Or it looks too cryptic already?